axelrod/result_set.py

```diff
@@ -45,6 +47,8 @@ def __init__(self, players, interactions, progress_bar=True):
             progress_bar : bool
                 Whether or not to create a progress bar which will be updated
         """
+        if game is not None:
+            self.game = game
```

Comment:
> Why would my suggestion not work?

I might be misunderstanding. Is your suggestion to replace the two lines:

```
        if game is not None:
              self.game = game
```

with the single line:

```
self.game = Game()
```

If that's the case then it would not allow us to pass a non default game as done in this test https://github.com/Axelrod-Python/Axelrod/pull/662/files#diff-fea0766a5edc2285adc3af49e295100cR176

(That's obviously a nonsense game but theoretically different games could be used to see effect of R, P, S, T on tournament etc...)